src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -23,15 +23,7 @@ public int Serialize(ref byte[] bytes, int offset, Span value, IFormatterResolve
                 len++;
             }
 
-            if (value.Tags != null)
-            {
-                len++;
-            }
-
-            if (value.Metrics != null)
-            {
-                len++;
-            }
+            len += 2; // Tags and metrics
```

Comment:
Does this mean we now always include the dictionaries even if they're empty? If so, we should make sure that this doesn't break the Agent (even older versions). It's possible that it doesn't handle empty dictionaries well.

Reply:
It does indeed. But I didn't consider that it could possibly break the agent.

Reply:
I tested with 7.16 and it works as expected.